### PR TITLE
CMOS-69: Component docs for Loki

### PR DESCRIPTION
**Pull request by princerachit**
Depends on https://github.com/couchbaselabs/observability/pull/93
Signed-off-by: Prince Rachit Sinha <psinha36@gmail.com>

### CODE REVIEW

File: .aspell.en.pws [hunk 1]
```diff
@@ -59,3 +59,4 @@ CPUs
 MiB
 RBAC
 YAML
+loki
```
Comment (patrick-stephens) [inline .aspell.en.pws:62]: Please use the capitalised version, `Loki`. We should not be mixing and matching so want to catch these. It's the name of a tool like Prometheus. I imagine it may even be allowed as the name of a popular Norse god.

File: docs/modules/ROOT/pages/component-loki.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+https://grafana.com/oss/loki/[Loki^] is the https://prometheus.io/[Prometheus^]-inspired open source, multi-tenant log aggregation system. Loki is used to process and store collected Couchbase logs.
+
+Couchbase cluster generates logs of various operations which is important for monitoring and investigation purposes. These logs need to be parsed in order to search and aggregate them based on keys. Fluent bit forwards the couchbase logs to loki which enables processing and storage of logs.
+
+image:component-loki.png[]
+
+== Get Started
+
+Loki is managed by the xref:architecture.adoc[Couchbase Monitoring Observability Stack]. Loki is enabled by default in the CMOS stack, however, fluent bit needs to be configured for the logs to reach loki. Depending upon the type of deployment you need to enable fluent bit.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-loki.adoc:16]: Capitalise Fluent Bit.

This likely wants to link to the other PR adding the FB deployment details once that is in.

File: docs/modules/ROOT/pages/component-loki.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
+= Loki
+
+[abstract]
+Loki can be used to process and store Couchbase logs.
+
+== Overview
+
+https://grafana.com/oss/loki/[Loki^] is the https://prometheus.io/[Prometheus^]-inspired open source, multi-tenant log aggregation system. Loki is used to process and store collected Couchbase logs.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-loki.adoc:8]: Typically we use a single line per sentence.
It makes it easier to review.

File: docs/modules/ROOT/pages/component-loki.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+Couchbase cluster generates logs of various operations which is important for monitoring and investigation purposes. These logs need to be parsed in order to search and aggregate them based on keys. Fluent bit forwards the couchbase logs to loki which enables processing and storage of logs.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-loki.adoc:10]: Again, split up the sentences.

Something needs to forward the logs, we recommend Fluent Bit and provide a method to deploy it.

Loki could also be used to store other logs by a customer potentially too. They may have their own instance already, or be using Grafana Cloud as well.

**Review by patrick-stephens — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/component-loki.adoc (hunk 4 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-loki.adoc:10]: >Loki could also be used to store other logs by a customer potentially too. They may have their own instance already, or be using Grafana Cloud as well.

Do we need to mention this? We have already answered the question, "Why we need Loki and how does that fit in with Cloud native approach"

File: docs/modules/ROOT/pages/component-loki.adoc (hunk 2 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-loki.adoc]: >This likely wants to link to the other PR adding the FB deployment details once that is in.
Yes, for now I am resolving the comment for now as that Pr is not in yet

File: docs/modules/ROOT/pages/component-loki.adoc (hunk 4 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-loki.adoc]: Addressed the rewording

File: .aspell.en.pws [hunk 5]
```diff
@@ -59,3 +59,4 @@ CPUs
 MiB
 RBAC
 YAML
+Loki
```
Comment (patrick-stephens) [inline .aspell.en.pws:62]: new line please

Comment (patrick-stephens) [reply .aspell.en.pws:62]: you also need to update the total count at the top

File: docs/modules/ROOT/pages/component-loki.adoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+Loki is used to process and store collected Couchbase logs.
+
+ifdef::env-github[]
+:imagesdir: https://github.com/couchbaselabs/observability/raw/main/docs/modules/ROOT/assets/images
+endif::[]
+image:component-loki.png[]
+
+Couchbase cluster generates logs of various operations which is important for monitoring and investigation purposes.
+Loki provides the option to parse the logs enabling search and aggregation based on keys.
+Something needs to forward the logs, we recommend Fluent Bit and provide a method to deploy it.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-loki.adoc:18]: Something sounds a bit weird, I was just using it as an example. Maybe reword "Logs are pushed to Loki by various tools, we recommend...."

File: docs/modules/ROOT/pages/component-loki.adoc [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+== Get Started
+
+Loki is managed by the xref:architecture.adoc[Couchbase Monitoring Observability Stack](CMOS).
+Loki is enabled by default in the CMOS, however, Fluent Bit needs to be configured for the logs to reach Loki.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-loki.adoc:23]: Link to the Fluent Bit page.

Comment (princerachit) [reply docs/modules/ROOT/pages/component-loki.adoc:23]: done

File: .aspell.en.pws (hunk 5 above)
Comment (princerachit) [reply .aspell.en.pws]: done

**Review by patrick-stephens — COMMENTED** [review]
This looks ready to submit to me. Although you're missing the navigation links but the diagram is updated.

**Review by patrick-stephens — APPROVED** [review]
